common/src/main/kotlin/io/homeassistant/companion/android/database/server/ServerConnectionInfo.kt

```diff
@@ -1,26 +1,40 @@
 package io.homeassistant.companion.android.database.server
 
-import android.Manifest
-import android.content.Context
-import android.content.pm.PackageManager
-import androidx.core.content.ContextCompat
 import androidx.room.ColumnInfo
 import androidx.room.Ignore
 import androidx.room.TypeConverter
-import io.homeassistant.companion.android.common.data.network.NetworkHelper
-import io.homeassistant.companion.android.common.data.network.WifiHelper
-import io.homeassistant.companion.android.common.util.DisabledLocationHandler
 import io.homeassistant.companion.android.common.util.kotlinJsonMapper
-import java.net.URL
+import io.homeassistant.companion.android.util.hasSameOrigin
 import kotlinx.serialization.SerializationException
+import okhttp3.HttpUrl
 import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
-import timber.log.Timber
-
-sealed interface SecurityStatus {
-    data object Secure : SecurityStatus
-    data class Insecure(val missingHomeSetup: Boolean, val missingLocation: Boolean) : SecurityStatus
-}
 
+/**
+ * Holds connection configuration for a Home Assistant server.
+ *
+ * This data class is embedded in [Server] and stored in Room database. It contains all URL
+ * endpoints and network detection settings used to connect to the server.
+ *
+ * @property externalUrl The primary URL used to connect to the server from outside the home network.
+ * @property internalUrl Optional URL used when the device is on the home network.
+ * @property cloudUrl Optional Home Assistant Cloud remote UI URL. When available and [useCloud] is enabled.
+ * @property webhookId Unique identifier assigned by Home Assistant during device registration.
+ * @property secret Shared secret for webhook authentication, if configured on the server.
+ * @property cloudhookUrl Direct webhook URL provided by Home Assistant Cloud.
+ * @property useCloud Whether to prefer the cloud URL over the external URL when not on the
+ *   home network.
+ * @property internalSsids List of Wi-Fi network names (SSIDs) that indicate the device is on
+ *   the home network.
+ * @property internalEthernet When `true`, Ethernet connections are treated as being on the home
+ *   network. `null` means the user hasn't configured this setting.
+ * @property internalVpn When `true`, VPN connections are treated as being on the home network.
+ *   `null` means the user hasn't configured this setting.
+ * @property prioritizeInternal When `true`, always try the internal URL first regardless of
+ *   detected network state.
+ * @property allowInsecureConnection Controls whether HTTP (non-HTTPS) connections are permitted
```

Comment:
Insecure is when not HTTPS. We should make sure to consistently check for it that way.

```suggestion
 * @property allowInsecureConnection Controls whether non-HTTPS connections are permitted
```